contracts/XVSVault/XVSVault.sol

```diff
@@ -4,18 +4,24 @@ pragma experimental ABIEncoderV2;
 import "../Utils/ECDSA.sol";
 import "../Utils/SafeBEP20.sol";
 import "../Utils/IBEP20.sol";
-import "./XVSVaultProxy.sol";
 import "./XVSVaultStorage.sol";
 import "./XVSVaultErrorReporter.sol";
 import "../Utils/SafeCast.sol";
+import "../Governance/AccessControlledV5.sol";
 
 interface IXVSStore {
     function safeRewardTransfer(address _token, address _to, uint256 _amount) external;
 
     function setRewardToken(address _tokenAddress, bool status) external;
 }
 
-contract XVSVault is XVSVaultStorage, ECDSA {
+interface IXVSVaultProxy {
+    function _acceptImplementation() external returns (uint);
+
+    function admin() external returns (address);
+}
+
+contract XVSVault is XVSVaultStorageV2, ECDSA, AccessControlledV5 {
```

Comment:
If we use multiple inheritance, we need a `__gap` in `XVSVaultStorage`, otherwise the further upgrades will overwrite the storage of `AccessControlledV5`.

As a side note, with gaps using versioned storage becomes kinda tricky, so maybe we can just add new entries (`vaultPaused` and `__gap`) to `XVSVaultStorage`? Not sure here.